test/stage-options.test.ts

```diff
@@ -187,13 +187,14 @@ describe('job settings', () => {
 
       pipeline.addStageWithGitHubOptions(stage, {
         jobSettings: {
-          if: 'github.repository == \'github/repo\'',
+          if: 'github.repository == \'github/repoB\'',
         },
       });
 
       app.synth();
 
-      expect(readFileSync(pipeline.workflowPath, 'utf-8')).toContain('if: github.repository == \'github/repo\'\n');
+      expect(readFileSync(pipeline.workflowPath, 'utf-8')).toContain('if: github.repository == \'another/repoA\'\n');
+      expect(readFileSync(pipeline.workflowPath, 'utf-8')).toContain('if: github.repository == \'github/repoB\'\n');
```

Comment:
can we snapshot test this one so that we can look at the resulting pipeline changes and reason through if they are the same?

Reply:
Sure.

Based on your other comment, it sounds like you'd prefer the `if` work to be in its own PR, yeah? How about this:
1. I'll back-out the `if` changes in this PR branch
2. Wait until this PR is merged
3. Then re-apply the `if` changes to a new branch off of the main branch and make a new issue+PR
4. Make this change there?

Reply:
Yep exactly. Looks like `if` changes may be more complicated than this PR lets on, and I'd rather focus on it on a separate PR.